spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/GqlDatastoreQuery.java

```diff
@@ -169,6 +157,19 @@ else if (this.queryMethod.isCollectionQuery()) {
 		return result;
 	}
 
+	private Object buildPageOrSlice(Object[] parameters, ParsedQueryWithTagsAndValues parsedQueryWithTagsAndValues,
+			DatastoreResultsIterable found) {
+		Pageable pageableParam =
+				new ParametersParameterAccessor(getQueryMethod().getParameters(), parameters).getPageable();
+		List resultsList = found == null ? Collections.emptyList()
+				: (List) StreamSupport.stream(found.spliterator(), false).collect(Collectors.toList());
```

Comment:
Why do we no longer need to `convertOnRead` and `applyProjection`?
I'm having a hard time following what the fix is conceptually.

Reply:
Perhaps a PR description would help? I am equally confused, BTW :)